server/src/test/java/org/elasticsearch/action/support/broadcast/unpromotable/TransportBroadcastUnpromotableActionTests.java

```diff
@@ -155,47 +155,52 @@ private static class TestBroadcastUnpromotableRequest extends BroadcastUnpromota
         }
     }
 
-    private static List<ShardRouting.Role> getReplicaRoles(int numPromotableReplicas, int numSearchReplicas) {
+    private static List<ShardRouting.Role> getReplicaRoles(int numPromotableReplicas, int numUnpromotableReplicas) {
         List<ShardRouting.Role> replicaRoles = Stream.concat(
-            Collections.nCopies(numPromotableReplicas, randomBoolean() ? ShardRouting.Role.DEFAULT : ShardRouting.Role.INDEX_ONLY).stream(),
-            Collections.nCopies(numSearchReplicas, ShardRouting.Role.SEARCH_ONLY).stream()
+            Collections.nCopies(
+                numPromotableReplicas,
+                randomBoolean() ? ShardRouting.Role.DEFAULT
+                    : randomBoolean() ? ShardRouting.Role.INDEX_ONLY
+                    : ShardRouting.Role.INDEX_SEARCH
```

Comment:
Maybe use `randomFrom()` instead? (Unless the skew towards `ShardRouting.Role.DEFAULT` is deliberate)